### DISCUSSION TITLE
Netbox 2..11.12 and python 3.7.5

### DISCUSSION THREAD

**Discussion by dejantep** [netbox-community/netbox#7704] [Q&A]
Hello,

I installed python 3.7 to prepare for going to netbox v. 3.0. This is now my default python3 on the system according to pyhton3 -V
I didnt now how to make sure netbox was using the new python version  so i decided to restart netbox and netbox-rq service and now netbox is down.

Is 2.11 note compatible with python 3.7 ?

**Comment by dejantep** [comment]
well i switched back to 3.6 using "update-alternatives" and everything is ok again. 
So i guess that this is not possible to do prior to upgrading to Netbox 3.0 ?

**Reply by candlerb** [reply]
Don't switch python version using update-alternatives (this will break other things on your system).

Instead, set the python version inside the Netbox virtualenv, by specifying it when you run upgrade.sh:

```
PYTHON=python3.7 ./upgrade.sh
```

This will rebuild your virtualenv with python pointing at python3.7.  You have to remember to do this each time you run upgrade.sh in future as well though.

I have a system which is Ubuntu 18.04 with python3.8 running this way.

**Reply by dejantep** [reply]
if i understand you correctly i could do your suggestion above with netbox 2.11.x and it will run fine?

**Reply by candlerb** [reply]
I am not sure.  The change to honour the `PYTHON` environment variable was made in commit 47ef8b9c dated Jul 19 2021, but it doesn't appear to have made it into v2.11.12.

However, I'd say there's no particular reason to upgrade python *before* moving to Netbox v3.0.
